vars/infra.groovy

```diff
@@ -142,7 +142,12 @@ Object withArtifactCachingProxy(boolean useArtifactCachingProxy = true, Closure
     echo "INFO: using artifact caching proxy from '${requestedProxyProvider}' provider."
     configFileProvider(
         [configFile(fileId: "artifact-caching-proxy-${requestedProxyProvider}", variable: 'MAVEN_SETTINGS')]) {
-          body()
+          withEnv([
+            "MAVEN_ARGS=-s $env.MAVEN_SETTINGS",
+            "ARTIFACT_CACHING_PROXY_ORIGIN=https://repo.${requestedProxyProvider}.jenkins.io"
```

Comment:
Just for fun, or do you expect something to use this?

Reply:
I've added this for https://github.com/jenkins-infra/javadoc/blob/master/Jenkinsfile#L35
(and other similar potential changes)

Not mandatory at all, I can keep my current Javadoc version.

Reply:
I see, can be used for tools not running Maven like https://github.com/jenkins-infra/javadoc/blob/595c5188e24434669bd522bfb1aefaa23cdab2be/scripts/generate-javadoc.sh#L21 :+1: